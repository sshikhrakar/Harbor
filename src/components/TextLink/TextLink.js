import React from 'react';
import PropTypes from 'prop-types';
import { TouchableOpacity, Text, StyleSheet } from 'react-native';

import colors, { DEFAULT_BUTTON_OPACITY } from '../../config/colors';

function TextLink(props) {

  const {
    onPress,
    disabled,
    text,
    style,
    activeOpacity,
  } = props;

  return (
    <TouchableOpacity
      onPress={ onPress }
      style={ styles.linkContainer }
      disabled={ disabled }
      activeOpacity={ activeOpacity }
    >
      <Text style={ [styles.link, style] }> { text } </Text>
    </TouchableOpacity>
  );

}

TextLink.propTypes = {
  disabled: PropTypes.bool,
  style: Text.propTypes.style,
  activeOpacity: PropTypes.number,
  text: PropTypes.string.isRequired,
  onPress: PropTypes.func.isRequired,
};

TextLink.defaultProps = {
  disabled: false,
  activeOpacity: DEFAULT_BUTTON_OPACITY,
};

const styles = StyleSheet.create({
  linkContainer: {
  },

  link: {
    color: colors.SOFT_BLUE,
  },
});

export default TextLink;
