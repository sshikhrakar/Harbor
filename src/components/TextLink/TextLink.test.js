import React from 'react';
import renderer from 'react-test-renderer';

import TextLink from './TextLink';

describe('COMPONENTS: TextLink', () => {

  it('should render correctly', () => {
    const tree = renderer.create(
      <TextLink
        text="SampleButtonText"
        onPress={ () => true }
        activeOpacity={ 0.3 }
        style={{ color: 'red' }}
      />
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });

});
