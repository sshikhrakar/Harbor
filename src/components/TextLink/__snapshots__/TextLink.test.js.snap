// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`COMPONENTS: TextLink should render correctly 1`] = `
<View
  accessibilityComponentType={undefined}
  accessibilityLabel={undefined}
  accessibilityTraits={undefined}
  accessible={true}
  collapsable={undefined}
  hitSlop={undefined}
  isTVSelectable={true}
  nativeID={undefined}
  onLayout={undefined}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
  testID={undefined}
  tvParallaxProperties={undefined}
>
  <Text
    accessible={true}
    allowFontScaling={true}
    disabled={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#3498db",
        },
        Object {
          "color": "red",
        },
      ]
    }
  >
     
    SampleButtonText
     
  </Text>
</View>
`;
